src/sys/unix.rs

```diff
@@ -537,24 +537,17 @@ impl SockAddr {
         docsrs,
         doc(cfg(all(feature = "all", any(target_os = "android", target_os = "linux"))))
     )]
-    pub fn vsock(cid: u32, port: u32) -> io::Result<SockAddr> {
-        unsafe {
-            SockAddr::init(|storage, len| {
-                // Safety: `SockAddr::init` zeros the address, which is a valid
-                // representation.
-                let storage: &mut libc::sockaddr_vm = unsafe { &mut *storage.cast() };
-                let len: &mut socklen_t = unsafe { &mut *len };
-
-                storage.svm_family = libc::AF_VSOCK as sa_family_t;
-                storage.svm_cid = cid;
-                storage.svm_port = port;
-
-                *len = mem::size_of::<libc::sockaddr_vm>() as socklen_t;
-
-                Ok(())
-            })
+    pub fn vsock(cid: u32, port: u32) -> SockAddr {
+        let mut storage = MaybeUninit::<sockaddr_storage>::zeroed();
```

Comment:
You can reduce the amount of unsafe by calling `assume_init` here.
```suggestion
        let mut storage = unsafe { MaybeUninit::<sockaddr_storage>::zeroed().assume_init() };
```
This is sound because all of its fields are valid when zeroed.